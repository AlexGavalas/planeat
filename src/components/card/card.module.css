.glass {
    position: relative;
    backdrop-filter: blur(2px);
    background-color: transparent;
}

.glass::before {
    content: '';
    position: absolute;
    inset: 0;
    z-index: -1;
    background-color: white;
    opacity: 0.25;
    backdrop-filter: blur(5px);
}
