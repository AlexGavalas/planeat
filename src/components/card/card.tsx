import { Card as MantineCard } from '@mantine/core';
import { type PropsWithChildren } from 'react';

import styles from './card.module.css';

export const Card = ({ children }: PropsWithChildren) => {
    return (
        <MantineCard className={styles.glass} shadow="sm">
            {children}
        </MantineCard>
    );
};
