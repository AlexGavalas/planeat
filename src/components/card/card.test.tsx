import { renderWithUser } from '~test/utils';

import { Card } from './card';

describe('<Card />', () => {
    it('renders with content', () => {
        const { container } = renderWithUser(<Card>Card content</Card>);

        expect(container).toMatchSnapshot();
    });
});
