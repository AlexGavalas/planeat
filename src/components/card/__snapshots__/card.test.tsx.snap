// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> renders with content 1`] = `
<div>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <div
    class="glass m-e615b15f mantine-Card-root m-1b7284a3 mantine-Paper-root"
    style="--paper-shadow: var(--mantine-shadow-sm);"
  >
    Card content
  </div>
</div>
`;
